src/message.js

```diff
@@ -721,39 +722,42 @@ async function createVerificationObject(signature, literalDataList, keys, date =
       break;
     }
   }
+
+  const isOnePassSignature = signature instanceof OnePassSignaturePacket;
+  const signaturePacketPromise = isOnePassSignature ? signature.correspondingSig : signature;
+  const isSignaturePacketAvailable = !util.isPromise(signaturePacketPromise);
   if (!primaryKey) {
     keyError = new Error(`Could not find signing key with key ID ${signature.issuerKeyID.toHex()}`);
   } else {
     try {
-      signingKey = await primaryKey.getSigningKey(signature.issuerKeyID, null, undefined, config);
+      // We need the signature creation time to check whether the key was expired at the time of signing.
+      // If the signature packet is not available yet, we delay the check after verifying the signature
+      signingKey = await primaryKey.getSigningKey(
```

Comment:
Could we, rather than sometimes calling `getSigningKey` twice, simply get the signing key packet from the `getKeys` call above (and store it there), and then always call `getSigningKey` below, to simplify the code (and hopefully not need `isPromise`, etc)?